app/common/metrics.py

```diff
@@ -17,6 +20,23 @@
     'serialized_build_time_seconds',
     'Total amount of time that would have been consumed by builds if all work was done serially')
 
+internal_errors = Counter(
+    'internal_errors',
+    'Total number of internal errors',
+    ['type'])
+
+
+class ErrorType(str, Enum):
+    AtomizerFailure = 'AtomizerFailure'
+    NetworkRequestFailure = 'NetworkRequestFailure'
+    PostBuildFailure = 'PostBuildFailure'
+    SetupBuildFailure = 'SetupBuildFailure'
+    SubjobWriteFailure = 'SubjobWriteFailure'
+    ZipFileCreationFailure = 'ZipFileCreationFailure'
+
+    def __str__(self):
+        return self.value
```

Comment:
Do we need this? If so, mind adding a comment about why? Inheriting from `str` should give this to us for free but it depends on what consuming code is doing. I guess it's needed if consuming code does `str(...)` anywhere:
```python
In [64]: class Test(str, Enum):
    ...:     apple = 'apple'
    ...:     black = 'black'
    ...:     crust = 'crust'

In [65]: 'hey {}'.format(Test.apple)
Out[65]: 'hey apple'

In [66]: str(Test.apple)
Out[66]: 'Test.apple'

In [67]: format(Test.apple)
Out[67]: 'apple'
```